src/ReactiveUI.Tests/MessageBusTest.cs

```diff
@@ -117,18 +110,18 @@ public void RegisteringSecondMessageSourceShouldMergeBothSources()
         }
 
         [Fact]
-        public void MessageBusThreadingTest()
+        public async Task MessageBusThreadingTest()
         {
             var mb = new MessageBus();
             int? listenedThread = null;
             int? otherThread = null;
             int thisThread = Thread.CurrentThread.ManagedThreadId;
 
-            Task.Run(() => {
+            await Task.Run(() => {
```

Comment:
Hmm, hadn't seen that, but does explain things! I think this was the only consistent deadlock, so hopefully we're not doing it elsewhere!